libstuff/STime.cpp

```diff
@@ -54,3 +54,30 @@ string SCURRENT_TIMESTAMP_MS() {
     snprintf(msString, 5, "%03lu", ms);
     return timestamp + "." + msString;
 }
+
+string SFirstOfNextMonth(string timeStamp) {
+
+    list<string> parts = SParseList(timeStamp, '-');
+
+    struct tm t = {0};  // Initalize to all 0's
+    t.tm_year = stoull(parts.front()) - 1900;  // This is year - 1900
+    parts.pop_front();
+
+    // Month values start at 0 in tm structs, so values are off by one.
+    uint64_t month = stoull(parts.front()) -1;
```

Comment:
Pretty sure that months `01` through `09` are going to be interpreted as octal if you don't specify base 10, which is great except for in August and September.